url2df0f7c811_przemekgalazka_legacycode_tgz-pJ8-pl_cc_core_cmd_CommandJ8/src/ccproxy/DBConnection.java

```diff
@@ -0,0 +1,63 @@
+package ccproxy;
+
+import java.sql.Connection;
+import java.sql.DriverManager;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+
+public class DBConnection {
+    Connection conn = null;
+
+    public DBConnection(String dbHost, String dbPort, String dbName, String dbUser, String dbPass) {
+
+        try {
+            Class.forName("org.postgresql.Driver");
+        } catch (ClassNotFoundException cnfe) {
+            cnfe.printStackTrace();
+            System.exit(1);
+        }
+
+        try {
+            String connString = new String("dbc:postgresql://" + dbHost + ":" + dbPort + "/"
+                    + dbName);
+            conn = DriverManager.getConnection(connString, dbUser, dbPass);
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**
This line of code assigns a resource to a class field and there is no method that closes or returns this resource. This can result in a resource leak that slows down or crashes your system.

**Fix**
Create a method in this class that closes the following resource: *conn*. After you use this resource, use this method to prevent a resource leak.

**More info**
[View resource management guidelines at oracle.com](https://www.oracle.com/technetwork/articles/java/trywithresources-401775.html) (external link).